aspnetcore/fundamentals/logging/loggermessage.md

```diff
@@ -0,0 +1,185 @@
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet3)]
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet7)]
+
+Note how the exception is passed to the delegate in `QuoteDeleteFailed`:
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet11)]
+
+In the Index page code-behind, a successful quote deletion calls the `QuoteDeleted` method on the logger. When a quote isn't found for deletion, an `ArgumentNullException` is thrown. The exception is trapped by the `try`&ndash;`catch` statement and logged by calling the `QuoteDeleteFailed` method on the logger in the `catch` block (*Pages/Index.cshtml.cs*):
+
+[!code-csharp[Main](loggermessage/sample/Pages/Index.cshtml.cs?name=snippet5&highlight=14,18)]
+
+When a quote is successfully deleted, inspect the app's console output:
+
+```console
+info: LoggerMessageSample.Pages.IndexModel[4]
+      => RequestId:0HL90M6E7PHK5:00000016 RequestPath:/ => /Index
+      Quote deleted (Quote = 'You can avoid reality, but you cannot avoid the consequences of avoiding reality. - Ayn Rand' Id = 1)
+```
+
+When quote deletion fails, inspect the app's console output. Note that the exception is included in the log message:
+
+```console
+fail: LoggerMessageSample.Pages.IndexModel[5]
+      => RequestId:0HL90M6E7PHK5:00000010 RequestPath:/ => /Index
+      Quote delete failed (Id = 999)
+System.ArgumentNullException: Value cannot be null.
+Parameter name: entity
+   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
+   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
+   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
+   at LoggerMessageSample.Pages.IndexModel.<OnPostDeleteQuoteAsync>d__14.MoveNext() in 
+      <PATH>\sample\Pages\Index.cshtml.cs:line 87
+```
+
+## Implementing LoggerMessage.DefineScope
+
+Define a [log scope](xref:fundamentals/logging/index#log-scopes) to apply to a series of log messages using the [DefineScope(String)](/dotnet/api/microsoft.extensions.logging.loggermessage.definescope) method.
+
+The sample app has a **Clear All** button for deleting all of the quotes in the database. The quotes are deleted by removing them one at a time. Each time a quote is deleted, the `QuoteDeleted` method is called on the logger. A log scope is added to these log messages.
+
+Enable `IncludeScopes` in the console logger options:
+
+[!code-csharp[Main](loggermessage/sample/Program.cs?name=snippet1&highlight=22)]
+
+Setting `IncludeScopes` is required in ASP.NET Core 2.0 apps to enable log scopes. Setting `IncludeScopes` via *appsettings* configuration files is a feature that's planned for the ASP.NET Core 2.1 release.
+
+The sample app clears other providers and adds filters to reduce the logging output. This makes it easier to see the sample's log messages that demonstrate `LoggerMessage` features
```

Comment:
Missing period after the 2nd sentence

Reply:
Periods .... soooo overrated. Who has the time to slow down?!! lol :smile: